### PR TITLE
planner, executor: IndexMerge supports reading extraHandleCol in partialTableReader

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23569 

Problem Summary:
PartialTableWorker for IndexMerge does not support reading extraHandleCol.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

N/A

### Check List 

Tests 

- Integration test

Side effects

N/A

### Release note 

- IndexMerge supports reading extraHandleCol in partialTableReader

### CODE REVIEW

File: executor/index_merge_reader_test.go [hunk 1]
```diff
@@ -78,3 +78,29 @@ func (s *testSuite1) TestIssue16910(c *C) {
...
+		PARTITION p202104 VALUES LESS THAN (1640966400),
+		PARTITION p202201 VALUES LESS THAN (1648742400),
+		PARTITION p202202 VALUES LESS THAN (1656604800),
+		PARTITION p202203 VALUES LESS THAN (1664553600),
+		PARTITION p202204 VALUES LESS THAN (1672502400),
+		PARTITION p202301 VALUES LESS THAN (1680278400)
+	);`)
+	tk.MustExec("insert tt value(1, 1577807000, 'jack'), (2, 1577809000, 'mike'), (3, 1585670500, 'right'), (4, 1601481500, 'hello');")
+	tk.MustExec("set @@tidb_enable_index_merge=true;")
+	tk.MustQuery("select count(*) from tt partition(p202003) where _tidb_rowid is null or (_tidb_rowid>=1 and _tidb_rowid<100);").Check(testkit.Rows("1"))
```
Comment (eurekaka) [inline executor/index_merge_reader_test.go:105]: Add a hint to force using index merge plan?

**Review by eurekaka — APPROVED** [review]
/LGTM

**Review by Reminiscent — APPROVED** [review]
LGTM

**Comment by qw4990** [comment]
@XuHuaiyu Friendly ping, should we go on to promote this PR?

**Pull request merged**